### DISCUSSION TITLE
【MQTT features】

### DISCUSSION THREAD

**Discussion by doujialong** [remibert/pycameresp#10] [Ideas]
Hello, thank you very much for your project. I'm currently using the FREENOVE_CAM_S3 firmware. From the code I see there are implementations related to MQTT but they are all commented out. May I ask when you plan to enable this feature? For example, sending motion detected images to an MQTT server rather than Pushover.

**Comment by remibert** [comment]
Hello,

The mqtt client or the broker works, they are simply disabled because I do not have personal use of them.

Uncomment in the main.py mqtt_client=True to have the client, you will need in this case
an mqtt broker (I did the validation with mosquitto, but others should be able to work).
It is necessary to configure in the web pages, the location of the broker, password etc.

You have to look in topic.py to obtain the name of the supported mqtt topics (note it is prefixed by the hostname).

The mqtt broker and client do not support SSL, like all the servers in this project, I never managed to get this to work, 
the internal ram requirements by SSL exceed the capacity of the internal ram of the ESP32 with the camera.

Look in the mqttsample.py, I made several examples to test with mosquitto.

**Reply by doujialong** [reply]
Okay, I will try it, thank you!
